extensions/blocks/dialogue/components/time-stamp-control.js

```diff
@@ -0,0 +1,119 @@
+/**
+ * WordPress dependencies
+ */
+import {
+	BaseControl,
+	__experimentalNumberControl as NumberControl,
+} from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+
+function checkValidValue( val, max ) {
+	if ( val > max ) {
+		return val = max;
+	} else if ( val < 0 ) {
+		return val = 0;
+	}
+
+	return val;
+}
+
+const timeStampMap = [ 'hour', 'min', 'sec' ];
+
+/**
+ * Helper function to parse and stringify the time stamp,
+ * in the HH:MM:SS format.
+ * `HH` is optional, meaning when it's empty,
+ * the string won't contain it, returning MM:SS.
+ *
+ * @param {object} typeValue - { type: value } pair value of the time stamp.
+ * @param {Array} smh - current [ second, menute, hour ] array values.
+ * @returns {string} HH:MM:SS format.
+ */
+function setTimeStampValue( typeValue, smh ) {
+	if ( smh.length <= 2 ) {
+		smh.unshift( '00' );
+	}
+
+	const type = Object.keys( typeValue )?.[ 0 ];
+	if ( ! type ) {
+		return smh.join( ':' );
+	}
+
+	let newValue = String( checkValidValue( typeValue[ type ], type === 'hour' ? 23 : 59 ) );
+
+	// Mask HH:MM:SS values.
+	if ( newValue?.length === 1 ) {
+		newValue = `0${ newValue }`;
+	} else if ( newValue?.length === 0 ) {
+		newValue = '00';
```

Comment:
Should we base this on the length of the audio or perhaps a block option? Otherwise, the timestamps will format oddly.

Reply:
sorry, what do you mean?

it applies a mask to the value. If the input is `7`, it will be replaced by `07`.

> ...base this on the length of the audio

Surely in a follow-up PR. It seems to be out of the scope of this implementation.